packages/react-graphql/src/links/operations-details-link.ts

```diff
@@ -0,0 +1,84 @@
+import {ApolloLink, Observable, FetchResult} from 'apollo-link';
+import {now} from '@shopify/performance';
+
+export interface GraphQLOperation {
+  name: string;
+  duration: number;
+  start: number;
+  end: number;
+  success: boolean;
+  errors?: {message: string; path?: string}[];
+  url?: string;
+  status?: number;
+  requestId?: string;
+}
+
+export interface Options {
+  requestIdHeader?: string;
```

Comment:
In web we used some custom header. Exposing this so the consumer can decide what request Id header they want to use.

Reply:
Actually, this header is simply `X-Request-ID`, should we just provide this header instead?

Reply:
Added `Header.RequestId` to `@shopify/network` and consumed it here